Doc/c-api/typeobj.rst

```diff
@@ -593,12 +593,6 @@ type objects) *must* have the :attr:`ob_size` field.
    ``Py_NotImplemented``, if another error occurred it must return ``NULL`` and
    set an exception condition.
 
-   .. note::
-
-      If you want to implement a type for which only a limited set of
-      comparisons makes sense (e.g. ``==`` and ``!=``, but not ``<`` and
-      friends), directly raise :exc:`TypeError` in the rich comparison function.
```

Comment:
Instead of removing this note, maybe you should change it say return `NotImplemented` instead.

Reply:
Unfortunately that clashes with guidance provided on the issue. :) Maybe this can get hashed out there?

Reply:
> Unfortunately that clashes with guidance provided on the issue

The important thing is that the bad advice (`raise TypeError`) is removed. The consensus is to `return NotImplemented` in this case. How explicit you want to make that in the docs is bikeshedding territory.

In other words: I'm totally fine with just removing that paragraph.